### ISSUE TITLE
Confusion about this line in model.py of Image Caption

### ISSUE THREAD

**Issue by x-zho14** [yunjey/pytorch-tutorial#93]
`

class EncoderCNN(nn.Module):

    def __init__(self, embed_size):
        super(EncoderCNN, self).__init__()
        resnet = models.resnet152(pretrained=True)
        modules = list(resnet.children())[:-1]      # delete the last fc layer.
        self.resnet = nn.Sequential(*modules)
        self.linear = nn.Linear(resnet.fc.in_features, embed_size)
        self.bn = nn.BatchNorm1d(embed_size, momentum=0.01)
        self.init_weights()
        
    def init_weights(self):
        """Initialize the weights."""
        self.linear.weight.data.normal_(0.0, 0.02)
        self.linear.bias.data.fill_(0)
        
    def forward(self, images):
        """Extract the image feature vectors."""
        features = self.resnet(images)
        features = Variable(features.data)
        features = features.view(features.size(0), -1)
        features = self.bn(self.linear(features))
        return features`
`features = Variable(features.data)`

What is the usage for this line? I think it has no change in features.
Thank you very much!!!

Zhou Xiao

**Comment by icfear** [comment]
Encoder input images are defined as Volatile in the following line.

https://github.com/yunjey/pytorch-tutorial/blob/6c785eb810848f259431865b12bb8795f4abcb23/tutorials/03-advanced/image_captioning/train.py#L62

Any variable with at least one volatile input will be volatile by default. Volatile variables also have requires_grad property set to False by default, which means that their gradient will not be computed. 
`features = Variable(features.data)` sets the volatile property to default value (false) so that requires_grad property could be transfered to the upper levels of the computational graph. You can check the following example for better understanding.

https://github.com/icfear/Pytorch-Example/blob/master/variable_example.py
